File: test_get_gradients.py
from keras.models import Sequential
from keras.layers import Dense, Activation
from keras import backend as k
import numpy as np
import tensorflow as tf


model = Sequential()
model.add(Dense(4, input_dim=2))
model.add(Dense(1, activation='sigmoid'))
model.compile(loss='binary_crossentropy', optimizer='adam')
model.summary()

outputTensor = model.output 
listOfVariableTensors = model.trainable_weights
gradients = k.gradients(outputTensor, listOfVariableTensors)


trainingExample = np.random.random((1,2))
sess = tf.InteractiveSession()
sess.run(tf.global_variables_initializer())
evaluated_gradients = sess.run(gradients,feed_dict={model.input:trainingExample})
evaluated_gradients

File: test_get_gradients.R
library(keras)
K <- keras::backend()
library(tensorflow)

model <- keras_model_sequential() %>%
  layer_dense(units = 4, input_shape = 2) %>% 
  layer_dense(units = 1, activation='sigmoid')
model %>% compile(loss='binary_crossentropy', optimizer='adam')
model %>% summary()

outputTensor <- model$output 
outputTensor
listOfVariableTensors <- model$trainable_weights
listOfVariableTensors
gradients <- K$gradients(outputTensor, listOfVariableTensors)
gradients

trainingExample <- rnorm(2) %>% matrix(nrow=1, ncol=2)
sess <- tf$InteractiveSession()
sess$run(tf$global_variables_initializer())

input <- model$input
input
evaluated_gradients <- sess$run(gradients,feed_dict = dict(input=trainingExample))
evaluated_gradients

